beacon-chain/rpc/eth/node/handlers.go

```diff
@@ -32,3 +56,363 @@ func (s *Server) GetSyncStatus(w http.ResponseWriter, r *http.Request) {
 	}
 	http2.WriteJson(w, response)
 }
+
+// GetIdentity retrieves data about the node's network presence.
+func (s *Server) GetIdentity(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "node.GetIdentity")
+	defer span.End()
+
+	peerId := s.PeerManager.PeerID().Pretty()
+
+	serializedEnr, err := p2p.SerializeENR(s.PeerManager.ENR())
+	if err != nil {
+		http2.HandleError(w, "Could not obtain enr: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	enr := "enr:" + serializedEnr
+
+	sourcep2p := s.PeerManager.Host().Addrs()
+	p2pAddresses := make([]string, len(sourcep2p))
+	for i := range sourcep2p {
+		p2pAddresses[i] = sourcep2p[i].String() + "/p2p/" + peerId
+	}
+
+	sourceDisc, err := s.PeerManager.DiscoveryAddresses()
+	if err != nil {
+		http2.HandleError(w, "Could not obtain discovery address: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	discoveryAddresses := make([]string, len(sourceDisc))
+	for i := range sourceDisc {
+		discoveryAddresses[i] = sourceDisc[i].String()
+	}
+
+	meta := &Metadata{
+		SeqNumber: strconv.FormatUint(s.MetadataProvider.MetadataSeq(), 10),
+		Attnets:   hexutil.Encode(s.MetadataProvider.Metadata().AttnetsBitfield()),
+	}
+
+	resp := &GetIdentityResponse{
+		Data: &Identity{
+			PeerId:             peerId,
+			Enr:                enr,
+			P2PAddresses:       p2pAddresses,
+			DiscoveryAddresses: discoveryAddresses,
+			Metadata:           meta,
+		},
+	}
```

Comment:
Is this clearer? Then we can remove `enr` and `meta`
```suggestion
	resp := &GetIdentityResponse{
		Data: &Identity{
			PeerId:             peerId,
			Enr:                "enr:" + serializedEnr,
			P2PAddresses:       p2pAddresses,
			DiscoveryAddresses: discoveryAddresses,
			Metadata: &Metadata{
				SeqNumber: strconv.FormatUint(s.MetadataProvider.MetadataSeq(), 10),
				Attnets:   hexutil.Encode(s.MetadataProvider.Metadata().AttnetsBitfield()),
			},
		},
	}
```